### ISSUE TITLE
Phraselists need an update

### ISSUE THREAD

**Issue by forid786** [e2guardian/e2guardian#264]
The phraselists included within E2Guardian are very old and can cause overblocking. I've created a pull request to disable "Japanese Pornography" phrase list which was in Alpha stage and was one of the biggest offenders of overblocking. And it was accepted and merged into the code. (Thanks Fred).

In my production environment, I've enabled MITM even for myself and have applied most of the restrictions which I place on everyone else. Just to see how well everything works, and to identify problems.
I've concluded the fact that the phraselists (and regex) definitely do need some updating and in many cases they over block. I understand that no filter is 100% accurate, but I've used Smoothwall Guardian at College which is essentially a fully rewritten DansGuardian with many of the same goals as E2Guardian but with commercial support. And due to the fact that it has phraselist updates quite frequently, I never even got close to the amount of false positives as E2Guardian, to be honest, barely any at all.

I tried visiting this site : https://community.talktalk.co.uk/t5/TalkTalk-TV/enable-IGMP-on-HG633-router/td-p/1725714

This is an UK based ISP forum, with nothing related to pornography. However it was detected by "content checking" as having "Pornography (German), Proxies, Pornography". On the page there really seems to be nothing specifically related to pornography or proxies.

I believe it's important that the phraselists are updated and paid a little more attention to, due to the fact that they effect one of the key functions of E2Guardian. Which is that dynamic blocking, based on content checking.

Hope this can get looked into, and improved! 👍 

EDIT: After scouring the web a bit, I've found some list(s) which maybe useful:
https://pastebin.com/gpHmA8X5
http://www.mediafire.com/?jb0cbe8aahc3lxd

**Comment by forid786** [comment]
Anyone?

**Comment by spikedrba** [comment]
@forid786 agreed there's an issue, that's also my experience however personally I find the idea of blacklists to be largely dead end to begin with unless they are managed by some org who is in that business and has therefore the resources and interest in getting that done right. I suppose there could be an effort, in the open source world, to create a "web filtering list", but even coordinating that effort would be a full time job. One example that seems to have work in a similar context is firehol "bad ips" list: http://iplists.firehol.org/ . Note that again, at the heart of this, there's organizations who have a very strong interest in building those lists to begin with.

My hope is that with v5 we'll be able to write plugins/additional modules and add support for external calls to web classification services such as zvelo. This is what many appliances do behind the scenes, including open source based solutions such as Untangle. I bet smoothwall does the same and that's one of the things you pay for (note even zvelo gets it wrong, it's hard to parse and make sense of the web!).

**Comment by forid786** [comment]
@spikedrba Smoothwall is actually surprisingly good at filtering. AFAIK smoothwall does still need to update its phraselists however I think we need to update the phraselists we've got on E2 Guardian somewhat before v5 release. Currently the way it is, there's a lot of false positives a bit too much, especially if you have MITM on.

**Comment by 1nfrag** [comment]
kang the phraselists off your smoothwall. Then upload them for us to use as your own work so we have no copyright issues. :)

**Comment by forid786** [comment]
@1nfrag I haven't got a Smoothwall box myself at the moment to even attempt to do that. On top of that from what I've read, smoothwall may have edited their phrase lists so it only works on their products (selfish much?)... Trying to rip it off may get us in trouble though, but I'm not too sure about that one.

**Comment by forid786** [comment]
Here's an example of why these things need to be updated: https://image.prntscr.com/image/MpuidAqXRtS6CI8FjlWzQA.png

A Googly Analytics link was detected as porn...

**Comment by forid786** [comment]
E2Guardian v5 is running amazing, I'm very happy with it! However we still need these old phraselists to be updated so people can get up and running quicker.

Anyone managed to get some fresh phraselists to keep this tool running great?

**Comment by Dalacor** [comment]
Yes I agree. When I first started Dansguardian/E2guardian in 2013, I used all the phraselists and weightedphraselist and ended up with horrible overblocking! I think the Japanese ones were particularly problematic if I remember rightly.

I ended up disabling all the Alpha Phraselists. I am worried that years later, things like this still have not been addressed. In some ways, despite all the releases, the program seems to be standing still as many bugs/issues raised years ago are still outstanding.

**Comment by forid786** [comment]
I also detected that the Japanese phrase list was crazy and actually sent a pull request to disable it by default.

Unfortunately, no one wants to provide the "brains" which is the phrase lists and keep them up to date. I seriously wouldn't even mind donating if need be. I know Philip must have some really up to date lists for E2BN but I guess he's keeping that separate as it's a paid service.

**Comment by Dalacor** [comment]
I think this is the problem with E2guardian. It needs to become a paid service to some extent so the developers can hire somebody to do these jobs because otherwise they will never ever get done. I don't have the time nor the expertise to do something like that.

The problem with Smoothwall etc is that they are crazy expensive.  I would not mind paying for the E2guardian service if it was reasonably priced and it fixed issues raised ten years ago!

By the way, did you see my post regarding reporting, which you responded to a few months back. Thanks for your input on that.

**Comment by philipianpearce** [comment]
OK, lets be clear, e2guardian is currently a open source filtering engine project.   The phraselists supplied are just samples of what can be done.   This was also true of dansguardian.  

Before we (at E2BN) first started using dansguardian for our customers we had to do lots of work to make the engine more robust, add new features and well as adjust the phrase lists so as to work effectively for our customers (who are mainly schools in England). We found that the sample phrase lists over blocked in some areas, but badly failed to block in others. So nothing new here!  We have always fed back our improvements to the source code to dansguardian/e2guardian community.   But as the phrase lists we use have (and are) being developed by constant feedback from our users we have kept these for our customers use. After all they are paying us to do this as well as paying for majority of the development costs of the e2guardian project!

However, I can see that it would be useful to the community to have more up-to-date phrase lists.  We will consider putting a single snapshot of our English phase lists into the e2guardian distribution, if other language users will do the same.  Our work is in English.  It requires a native speaker to develop and maintain a good phrase list in a language.  This should supply a robust sample set of phrase lists for users to use at home or for test purposes in the work place or school.

We are also considering providing a paid list (url and phrase) update service based on our Protex lists for e2guardian users.  This would provide regular updates to both lists and phrase lists.  Is this of interest to anyone in the community?   How much would you be prepared to pay?   And any suggestions on how would we ensure that this service is not abused by other commercial companies using e2g in their paid service? 

@Dalacor If your requirement is a packaged, English speaking, full filtering service then E2BN Protex would be happy to give you a quote for your requirements which is reasonably priced and includes the use of IWF and PROTECT lists which are must haves for UK schools. See https://www.e2bn.org/cms/web-filtering/protext-web-filtering

**Comment by forid786** [comment]
@Philipianpearce I'd love to see a snapshot of the English phrase lists in E2 Guardian!! 

I mainly use E2 Guardian for home, as long as the paid list is reasonably priced and affordable. I do not mind supporting your work at all! The price I'm prepared to pay depends on how you'd like to model your system. Monthly payments? Yearly?

This may apply to others that are in organisations that have a smaller IT budget and cannot shell out a lot of money for like your E2BN Protex solutions or Smoothwall etc. Therefore I think a fair and affordable price would be very key!

One way you can stop abuse is by encrypting the paid lists and although some maybe against the idea. You may need to include a proprietary module to accommodate the paid list. 

I'd be very interested to hear other people's input, the above are just my opinions.

PS: Apart from updated lists how different is the E2BN system compared to E2 Guardian? I know you're making use of the code from E2 Guardian but what specific areas are there where E2BN is better apart from the lists?

**Comment by Dalacor** [comment]
I concur with @forid786 that it would be very useful to understand the differences between E2Bn and E2guardian. I have looked at your link and I think that E2guardian can do virtually everything that E2BN so I suspect the differences are more gui based - i.e. easier to change settings?

I have just in the last month purchased a blacklist subscription from squidblacklist.org. They charge $275 for three years ($92 a year). I would recommend that you use that as a basis for a price range. They only supply blacklists, not phraselists though, so that would affect the price a bit. They use a username/password to protect their blacklists.

> And any suggestions on how would we ensure that this service is not abused by other commercial companies using e2g in their paid service?

I assume that you mean that you want to offer one price for home users and another price for business users like myself using E2guardian in schools?

Around Xmas, I will review E2BN as compared to E2guardian, but I have just renewed my clients contracts recently based on current pricing, so won't move to E2BN until April next year. Having said that, I am quite happy with E2guardian on the whole. I only have two real issues with it - Documentation and a sense that some bugs have been outstanding for years.

With regards to E2guardian being open source engine product or paid list or donation based or moving to subscription based I don't know what other people using E2guardian are using E2guardian for, so I can't speak for other people.

Speaking for myself, I would be happy to support E2guardian financially and providing bug reports, assisting with documentation and providing feedback if this would improve the following issues:

**Documentation** - see my guide for example (please note that this guide is out of date and was written primarily for myself but also to give other E2guardian users a rough guide) - https://www.dalacor.co.uk/e2guardian.php. This guide will be updated in the next few weeks when I update to version 5 E2guardian, but you can get an idea of what I mean by documentation. You will notice in that guide for example that I enabled --enable-sslmitm=yes and then removed it as I could not get it to work. At the time there was not much documentation on how to get MITM to work on that version.

Another example of where documentation is important is - the E2guardian files default to a very low children count and the websites talking about Dansguardian info is very outdated as to what is suitable for a modern Internet. I have used the dstats.log to help me tweak this at one school, but I found that log file by accident recently. It would have been very useful years ago when setting up E2guardian.

maxchildren = 300
minchildren = 150
maxsparechildren = 150
maxagechildren = 800

**Updated Phraselists and updated files** like bannedsitelist using blacklists such as squidblacklist. Urlblacklist is no longer developed, yet the bannedsitelist file still points to a configuration using UrlBlacklist.

Also it would help to fix bugs like this one - authplugin = '/usr/local/e2guardian/etc/e2guardian/authplugins/port.conf' is still missing in version 5 of the e2guardian.conf file.

I hope that my feedback helps. I think open source is great, but this topic opened up by the OP highlights one of the drawbacks with opensource. I would look into providing some sort of a subscription service for those of us who want to contribute to E2guardian to improve the product bearing in mind some of us are home users and others are companies.

Thank you.

**Comment by forid786** [comment]
@Dalacor Squid black lists are now free, did you check the Google groups page for E2 Guardian?

I tried it out and ended up sticking to Shalla List for my black list. The categorisation and websites better suited my application. 

I'd like E2 Guardian to stay open source so that everyone can contribute to make it better and I'm even hoping that one day we can get it into pfSense's official repository. With than extra paid module for the updated phrase lists. As a whole, pfSense is a great firewall but it's not very great for filtering so I use E2 Guardian on it. Then once E2 Guardian makes it into a "channel" if you will and more people start using it. I'm sure we'd get more traction in all directions from code contributions, to paid phrase list users etc.

**Comment by Dalacor** [comment]
I think that you are referring to another Squid Blacklist. I am talking about this site - http://www.squidblacklist.org/. Maybe you are talking about this - https://blog.squidblacklist.org/?p=1658 - that is malicious domains and IP Blacklists are free - not all the blacklists - just the malicious.

I didn't go for Shall based on this - https://blog.squidblacklist.org/?p=1424 - Granted I have only their word for it, but I did some testing with their Squid blacklists malicious domains and found that on most emails with a malware link, squid blacklist had that domain on their malicious blacklist.

Yes I understand the logic of having E2 Guardian staying open source. I am just trying to find ways to improve what I see as the two biggest problems with E2guardian - documentation and bug fixes (including out of date things like phraselists. I am not sure what the solution is, I was just throwing out some ideas.

I think a free module for the basic program and a paid for module for extras makes sense.

**Comment by forid786** [comment]
@Dalacor You're correct, I assumed it was the same. Squid Blacklist blog is correct, Shalla List isn't the best and I've tested a lot of the domains myself and found a lot of parked domains and whatnot. But for the most part it does the job, and has most of the porn domains on there. The rest get picked up by E2 Guardian phrase lists.

For home usage to be honest, I haven't found the need to get anything better. I may look into it if it gives me any problems as until recently I was using Open DNS as well but CloudFlare managed to win my heart. 

Let me know if there's anything else you use which could be useful! :)

I'm still waiting on @Philipianpearce if the E2 Guardian team manage to whip together a great list / phrase list at an affordable price. I'd go with that instead. 

Sorry for the late response by the way, for some reason GitHub didn't send a notification.

**Comment by Dalacor** [comment]
No need to apologise for the late response. Sometimes I get so busy that I don't respond for a couple of weeks to any reply here!

Yes, your needs are different from mine so I have no doubt that Shall is suitable for your needs. Glad that you tested it and confirmed this. I never actually got around to testing this as I assumed that they were telling the truth on the basis it was so easy to prove otherwise!

My problem is not phraselists - my biggest issue with filtering is google images! This is something that I want to fix over the August holidays.

What are you using Cloudflare for? Website Hosting?

@philipianpearce Sawmill is a company that provides universal log file analysis and reporting. They have a premium support package as well as the basic package (where you just buy the product).

Perhaps what you could do is have a addon package to buy subscription based things like phraselists and such like specific to E2guardian and have a premium support package which provides end user support for those that want support on issues where its too difficult to address on a forum like this. Just a suggestion.

**Comment by forid786** [comment]
@Dalacor I'm using CloudFlare for DNS at the moment as it is noticeably faster, it's noticeable more on sites which load a lot of external content. I have an internal DNS caching anyways via pfSense, but CloudFlare makes looking up new domains quicker.

If phrase lists/black list was up to date, would it not pick up Google images links? There are group options for enabling the scanning of links within a page.

@philipianpearce It's been sometime since we discussed this, have you made any progress, got any new ideas on how we can implement this? E2 Guardian for me now works as I would like it to, but I need decent phrase lists / black list to provide the intelligence behind the system. 

PS: If anyone has any phrase lists/black lists they can share, PM me!

**Comment by Dalacor** [comment]
@forid786 

> If phrase lists/black list was up to date, would it not pick up Google images links? There are group options for enabling the scanning of links within a page.

I have now installed E2guardian 5.2.2 or whatever the latest version is as well as MITM. I have deepurlanalysis on (but I believe that this only affects images from blocked websites) but I have noticed that if I do a google search like free porn, I will see all the Google Search Results for porn websites. When I click on the links they are all blocked which means that the porn blacklist is working well. However, I would like to remove search results that are inappropriate in the same way that Google Safe Search Mode does. I should not see any of these website links at all.

I have found that I can edit the bannedsearchlist file and put in terms like free+porn and it will block that search term (now that I have MITM working yay), so this will be useful for search terms where Google Safe Search Mode fails especially with Google Images.

Obviously I intend to have Google Safe Search enabled, but I would like in addition to be able to remove search results for websites that are either blacklisted or fail weighted phrase lists etc. Can this not be done? Is the deepurlanalysis only for webpages for static content and not for things like google search results?

**Comment by forid786** [comment]
@Dalacor I believe the issue here is the fact that Google, Yandex and a few other sites sort of proxy the image. So E2 Guardian itself would have to be smarter and have algorithms for checking the links on these images too.

Good news is, it's possible to do. And I know other products do it but E2 Guardian is a free open source offering. Things are taking longer to get done due to mainly Fred and Philip maintaining the project. But we're slowly getting there. So yeah, for the time being banned search list is what I'd recommend.

**Comment by Dalacor** [comment]
Sorry I am not talking about the google images here in my question. What I was asking about is if you go onto google and type say free porn and click enter, you will get a list of web results to porn websites. Not images. When I click on the search results, the websites come up as blocked which is good. But I was wondering if the deepurlanalysis was supposed to remove search results pointing to blocked websites?

Google Images I will handle by using Bannedsearchlist file which again I am also intending to use for actual Gooogle Search as well obviously as it affects both google search and google image search.

**Comment by forid786** [comment]
@Dalacor I did test this a while back and if MITM was enabled my phrase lists would block that Google search. I haven't tested very recently though.

**Comment by Dalacor** [comment]
Sorry @forid786. I am confusing you.

I am not talking about Phraselists or bannedsearch terms etc. Naturally if you have free porn in the banned phraselists file it will block that.

What I am actually asking is for example I might type in spring break (in other words school holidays) and it will give me say ten websites on the Google Search page.

Nine of those websites are relevant to school holiday activities but the tenth website link might be related to Spring Breakers (a movie) and that particular website is in the blacklists. Naturally if I click on the link it will say banned site Porn or whatever. However, I was wondering if there was a way to prevent Google Search from returning that 10th website as a search result not because the term spring break is being filtered, but because that link points to a banned website.

Hope that makes sense.

**Comment by forid786** [comment]
@Dalacor Alright now that makes perfect sense. It would be possible to modify the search queries as long as you MITM but it would probably require a lot of CPU resources and I'm not sure how much of a hit it would take to performance. But... If you have good hardware I guess it could be achieved with some modifications to E2 Guardian.

**Comment by benjaminnichols** [comment]
Hey everybody I have had to overcome the same problem with a lack of meaningful keywords, and I know that I am late to the table on this discussion, but I would be happy to contribute to your efforts here creating meaningful phrase lists data. 

Who do I email? 

Or just drop me a line at b@squidblacklist.org Ill be happy to send you some files with category specific regex and plain text queries which may prove quite helpful. 


As far as image classification goes, I believe there is currently a demand, a market, and a need to create a low cost or free API service,, with a high precision trained image classification driven database behind it, one that that administrators could leverage to create/enable something like a simple plugin to varous webfiltering platforms and applications, such as e2guardian. 

( psst any entrepreneurs good with TensorFlow reading this? )

That would enable multiple free or paid platforms to use the service, without the need for an expensive on site big box solution.

**Comment by Dalacor** [comment]
Hi Ben, I currently subscribe to your 3 year blacklist plan.

I would be pleased if you could include a Phrase list service as well as I believe that this would be more effective than a blacklist as most people find websites by searching in Google. So I would definitely be interested in the regex and plain text queries that you are talking about especially now that I am currently testing E2g5 with MITM and a big focus of my tests will be on search results in Google with and without safe mode.

From my usage of E2G, I think the main issue is that even with Google Safe Search kids can type in (mild porn related search terms) and come up with Image results. You can split this into two categories.

Phrase lists in E2G drastically need an update - which is something that you can probably help with. currently it both overblocks and underblocks - i.e. doesn't block things it should, and it blocks pages that should never be blocked. Phrase lists file is the biggest issue to the point that its not really being used in E2guardian because it doesn't work.

Google Images that are not appropriate still show up in search results (for both naughty and innocent searches) so something that scan images and determine whether there is nudity etc is probably what needs to be implemented. This is something that is probably more E2G related.

Image Classification - i.e. relying on wesites to categorise images doesn't work as most websites don't bother. E2G has actually removed the Pic content rating configuration probably because websites don't categorise their pictures. So unless you can create a database that can categorise pictures, I don't know how that would work.

Blacklists are vital, but phraselists would prevent a lot of bad websites from ever being listed in the first place as most people find websites through Google Search. So I would expend effort on Phrase Lists, not image classification etc.

**Comment by Dalacor** [comment]
@forid786 I presume that you are in touch with your developer that develops pfsense and also you seem to have an understanding of E2guardian that exceeds my understanding. So I hope that you will be able to advise on this issue.

My clients are currently running ver 3.1 or something like that. I am planning to upgrade to latest version in two weeks time. In addition to other issues, one of the main problems that we experience once in a blue moon is that I can only set the max children to be around 400. Mosf of the time this is sufficient, but we occassionally run out of child processes. E2guardian manages a lot more than 400 child processes, but the problem is that either Squid or Slackware essentially crashes because I run out of file descriptors so I cannot raise this limit. I have never been able to fix this issue. I am getting the errors in the Squid log, so I presume it's the limit on Squid that is the problem not the limit on Slackware?

If I have understood it correctly, I need to change this -  #define __FD_SETSIZE 1024 in high value like 8196 in the following two files below (Do not remove the # in front of line – just change the actual number)
/usr/include/bits/typesizes.h
/usr/include/linux/posix_types.h

and I need to do this - And put in e2guardian start script ulimit -n 8196, now you can increase maxchildren to 8176 = 8196 – 20 (reserved) max. What do I do here – this has never been set!

My first question - does the e2guardian startup script ulimit -n 8196 still need to be set in E2guardian 5.2 if using httpworkers? There is no documentation on this.

My second question - is there a limit on Squid causing the file descriptor issue or is the issue because I have not correctly set the typesizes.h and posix_types.h to be 8196 that is causing the running out of filedescriptors issue. If it's squid, I don't care about it, because I am removing Squid in the upgrade and using E2guardian only - no squid. Having said that, I have never set E2guardian to be more than 1000 child workers so its probably squid that is the issue.

My third question - do I need to increase the file descriptor sizes in typesizes.h etc for E2guardian 5.2 httpworkers or is this now irrelevant. The default size is 1024

Last question - I can't find documentation on how to calculate the correct number of httpworkers in 5.2 per user/website to work out how many httpworkers I actually need.

**Comment by shivanikhosa** [comment]
Hi All,
I'm in urgent need of a reliable blacklist source. urlblacklists no longer exit, dansguardian isn't being maintained. I was using squidblacklist with a yearly contract but sorry to see that they have reduced their lists considerably. Porn category has gone down from a 1 million to 7000 and total size has gone down from 3 million domains to like 300k. Can anyone please suggest an alternate source for blacklists which is maintained and reliable.

**Comment by Dalacor** [comment]
@shivanikhosa I also use Squidblacklist. It appears that the owner of Squidblacklist died in March. I suspect that his service had subscribed to other services and because he has passed away those subscriptions have fallen away. It would appear that his family are not maintaining the subscription blacklists so we will have to regard Squidblacklist as dead unfortunately. I have also noticed how the number of domains declined.

This is the one that I now use.  https://dsi.ut-capitole.fr/blacklists/index_en.php

I am planning to take the Squidblacklist and Url Blacklists and remove the dead domains, then make a file diffs between them and the Ut1 blacklists and then see if I can send the additions to the Ut1 blacklists maintainers. Subject to any legal issues of course. But as neither URL Blacklists nor Squid Blacklists effectively exist anymore, I can't see that there will be a problem.

This will ensure that we have the beneftis of all three blacklists as the Ut1 blacklists don't have as much as the Squidblacklists for some categories.

You should not be using Dansguardian as it hasn't been updated in more than 5 years. Only use E2guardian. However E2guardian is a filtering program not a blacklist so don't use it as a blacklist.

Having said that, the phrase lists inE2guardian do need an overhaul as the current ones overblock and underblock in that legitimate websites get blocked whereas things that should be blocked are not.

**Comment by shivanikhosa** [comment]
@Dalacor: Thanks for letting me know. Sad to hear about Ben Nichols's untimely demise.
Thanks for your pointers. I'll take a look at these. Have you used Shallalist (http://www.shallalist.de/) before ?

**Comment by Dalacor** [comment]
@shivanikhosa Shallalist as far as I am aware has not been updated since 2016. There are a lot of dead domains in Shallalist and they are not blocking everything that they should be. I don't even bother with Shallalist.

UT1 is not perfect but it actually seems to be blocking more than other blacklsts. For example there was a gaming website - UT1 is the only blacklist that blocks it. Draytek router blacklists, URL Blacklist, Squid Blacklist etc all let this website through. I also tested a few other websites. There are gaps, but I think if I can merge all the blacklists, this should solve most of the gaps.

**Comment by philipianpearce** [comment]
Sorry this has taken so long, but I have just uploaded to v5.3.dev a snapshot of updated phraselists.  These have been tuned by E2BN Protex Ltd so that false positives are greatly reduced for English language sites.

If you have more up-to-date phaselists in other languages or improvements to the revised English phrase lists, then please contribute them, so that we can improve the product overall.

Regards
Philip

**Comment by Dalacor** [comment]
Thank you. Very much appreciated. I will try and find the time this weekend to test this out as this will make a huge difference to the efficiency of the program. However, there is a lot going on at the moment outside the broadband filtering arena.

**Comment by forid786** [comment]
Thanks so much for this @philipianpearce ! 

Been waiting patiently, to be honest. Started to think you forgot. Thanks for the hard work! I will close this issue off and will test the new phraselists when I get a chance.

**Comment by Dalacor** [comment]
It is probably me not understanding how Github handles stable and development versions. When I download either 5.3dev or 5.3dev-bsd or 5.4dev and install using the ./autogen.sh ./configure make, make install and make clean method it installs E2guardian but does not install the new phraselists folder in E2guardian and presumably is missing other things.

I can see in the source folders that I have downloaded that the new phraselists folders are there, but when I compile E2guardian the new phraselists folders are missing. Also I get a lot of errors of missing files/folders etc in E2guardian 5.4dev which leads me to believe that the e2guardian install routine is only installing the E2G 5.3.3 stable version folders and files. I guess that I need to merge something somewhere?

Normally when I download the latest stable e2guardian, I download the tarball. With the development I only have the option to clone or download the zip version.

@forid786 did you get the new phraselists working? What is the procedure to install a development version instead of stable? I am trying to install a clean version of E2guardian v 5.4 if possible.

**Comment by Dalacor** [comment]
**Update**. I managed to find the e2guardian instructions git clone url, git branch -a and git checkout v5.4.dev but while it seems to install the e2guardian 5.4 version according to the e2guardian version in the installed /sbin directory, it's not installing the new phraselists and some of the other files, that 5.4 requires.

So I have just given up and installed the stable 5.3.3 version and will just have to wait until the 5.4 stable version comes out unfortunately. Hopefully that will be soon as I am really looking forward to trying out the new phraselists.

**Comment by philipianpearce** [comment]
@Dalacor  
As **v5.3.dev** is stable release track (so no breaking changes to config etc), I have simply added the new phraselists to the distribution, so they can be deployed if required.  The default config files still include the original lists and so to use the new lists, you need to manually copy the files into your working directory and then edit the the includes to point at the revised phraselists.

**V5.4.dev** is work in progress (not even yet a pre-release release) and so aspects such as packaging (e.g. make install, documentation) are not yet complete.    In the meantime to test the new phrase lists, after make install copy the directory structure in ./configs/lists/phraselists in the distribution to your working directory.   In this branch the sample master phraselist files point at the new lists and the old phraselists are provided as an option.

I am re-opening issue and will update you, once the make install work has been done.

**Comment by Dalacor** [comment]
Do you have an ETA on the release date for V5.4. After all the trouble I had this weekend trying to get the dev version installed, I just thought it might be safer to wait until the stable release is out as I don't want to put on an unstable version on a client network.

But from what you say it looks like V5.3.dev is stable. But if you are planning on releasing the v5.4 release soon, I can wait. I was just wanting to take advantage of the new phraselists. Thank you.

**Comment by forid786** [comment]
@Dalacor For the time being, I just went through the git clone method and just dropped the phrase lists in and then activated them in my config. It appears to be working much better, however I am still sort of testing this in my lab, and am not using it in production yet until I get a few kinks worked out such as cleaning up the older phraselists. Haven't had much time to play with it, but it's nice to be able to use phrase filtering again as a catch all for sites that are not in my blacklist. 

If 5.4 is coming out soon, it maybe worth waiting for @philipianpearce to go ahead and do the small bits of cleanup for you, and replace the English Phraselists with the ones provided by E2BN. 5.4 also should be quite good for me as I run E2Guardian in pfSense which is BSD based. I can see a lot of improvements and changes geared towards BSD.

**Comment by Dalacor** [comment]
@forid786 Yes I noticed that there was a bsd specific version as well interestingly enough. I am inclined to wait till 5.4 then as it should be released imminently.
